### DISCUSSION TITLE
Updating Supplier Price Breaks via API

### DISCUSSION THREAD

**Discussion by Fluxus00** [inventree/InvenTree#3990] [Q&A] [answered]
Hello,

i am currently trying to update some Price Breaks of SupplierParts via the API. As far as I understand the documentation it should work like this:

variable = SupplierPriceBreak.update(api, {
 "values": ....
})
Apparently this do not work, and i do not know why. My latest thoughts were, that the URL might be changed a little bit/ need more informations, or that the "update" has to be replaced by anything else.

If anyone could give me an example how to do it, I would really appreciate it.

**Comment by SchrodingersGat** [comment, marked as answer]
https://docs.inventree.org/en/latest/api/python/examples/#updating-attributes

**Reply by Fluxus00** [reply]
Okay thank you i can see what my mistake is. But can I ask you a question? Why is the method called "save",  does the original version of the part, pricebreak, whatever get "saved" somewhere or anything similar to that?

**Reply by matmair** [reply]
@Fluxus00 it saves the data you pass it on, therefore it is called saved. You could have an add and update function separately but this handles everything in one call.

**Reply by Fluxus00** [reply]
@matmair Alright, thank you both for the answers!

**Reply by matmair** [reply]
@Fluxus00 No worries, feel free to ask if anything else comes up!

**Comment by SergeoLacruz** [comment]
Hi,
For those of you who do not like to read through the code I started a small reference docu:

https://github.com/SergeoLacruz/inventree-docs/blob/master/docs/api/python/api_reference.md

It makes things easier for me when I need to look up certain things. Far from beeing complete.

**Reply by SchrodingersGat** [reply]
@SergeoLacruz if you can contribute any docs back upstream when they are ready, that would be greatly appreciated :)
